### PR TITLE
Test/misc/edit button visibility

### PR DESCRIPTION
**Pull request by francelu**
# Enhance UI Tests and Update Top Bar Titles

## Description
This PR enhances UI tests across multiple screens by adding assertions for component visibility and text content, and updates top bar titles for consistency. It closes issue #126.

### Enhancements to UI Tests
- Added assertions for component visibility and text content in the following test classes:
  - `AlertListScreenTest`
  - `AlertScreenTest`
  - `MapScreenTest`
  - `CreateProfileTest`
  - `ProfileScreenTest`
  - Introduced `TimerScreenTest` to check components visibility.

### Updates to Top Bar Titles
- Changed titles to ensure consistency:
  - "Alert Lists" for `AlertsList`
  - "Map" for `Map`
  - "Create Your Account" for `CreateProfile`
  - "Your Profile" for `ProfileScreen`
  - "Tampon Timer" for `TimerScreen`

#### Added
- `app/src/androidTest/java/com/android/periodpals/ui/timer/TimerScreenTest.kt`

#### Modified
- `app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt`
- `app/src/android/periodpals/ui/alert/AlertScreenTest.kt`
- `app/src/android/periodpals/ui/map/MapScreenTest.kt`
- `app/src/android/periodpals/ui/profile/CreateProfileTest.kt`
- `app/src/android/periodpals/ui/profile/ProfileScreenTest.kt`
- `app/src/main/java/com/android/periodpals/ui/alert/AlertsList.kt`
- `app/src/main/java/com/android/periodpals/ui/map/Map.kt`
- `app/src/main/java

## Tests
- Can't check code coverage in Sonarcloud :/

### CODE REVIEW

**Review by coaguila — APPROVED** [review]
Lgtm

File: app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/alert/AlertListScreenTest.kt:1]: Nice job finding this inconsistency!

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt [hunk 1]
```diff
@@ -2,25 +2,51 @@ package com.android.periodpals.ui.map
...
 class MapScreenTest {
 
+  private lateinit var navigationActions: NavigationActions
   @get:Rule val composeTestRule = createComposeRule()
 
+  @Before
+  fun setUp() {
+    navigationActions = mock(NavigationActions::class.java)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Route.ALERT)
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:30]:
```suggestion
    `when`(navigationActions.currentRoute()).thenReturn(Route.MAP)
```

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt [hunk 2]
```diff
@@ -2,25 +2,51 @@ package com.android.periodpals.ui.map
 
 // UI test for MapViewContainer.kt
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:3]: this comment is redundant, can you remove it?

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt [hunk 3]
```diff
@@ -2,25 +2,51 @@ package com.android.periodpals.ui.map
...
   fun testMapScreenWithPermissionGranted() {
     composeTestRule.setContent {
       PeriodPalsAppTheme {
-        MapScreen(
-            locationPermissionGranted = true,
-            navigationActions = NavigationActions(rememberNavController()))
+        MapScreen(locationPermissionGranted = true, navigationActions = navigationActions)
       }
     }
     // Verify that the map is displayed when permission is granted
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:52]: This comment could be removed if renaming the test `checkMapIsDisplayedWhenPermissionGranted`
This makes the code self-documenting and the tests more explicit

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt [hunk 4]
```diff
@@ -31,9 +57,7 @@ class MapScreenTest {
   fun testMapScreenWithoutPermission() {
     composeTestRule.setContent {
       PeriodPalsAppTheme {
-        MapScreen(
-            locationPermissionGranted = false,
-            navigationActions = NavigationActions(rememberNavController()))
+        MapScreen(locationPermissionGranted = false, navigationActions = navigationActions)
       }
     }
     // Verify that the map is still displayed even if permission is not granted
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:63]: this comment could be removed if renaming the test `checkMapIsDisplayedWhenNoPermissions`
As before, this makes the code self-documenting

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt [hunk 5]
```diff
@@ -2,25 +2,51 @@ package com.android.periodpals.ui.map
...
+    composeTestRule.onNodeWithTag(TopAppBar.TOP_BAR).assertIsDisplayed()
+    composeTestRule.onNodeWithTag(TopAppBar.TITLE_TEXT).assertIsDisplayed().assertTextEquals("Map")
+    composeTestRule.onNodeWithTag(TopAppBar.GO_BACK_BUTTON).assertIsNotDisplayed()
+    composeTestRule.onNodeWithTag(TopAppBar.EDIT_BUTTON).assertIsNotDisplayed()
+  }
+
   @Test
   fun testMapScreenWithPermissionGranted() {
     composeTestRule.setContent {
       PeriodPalsAppTheme {
```
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:48]: The call to `PeriodPalsAppTheme` is redundant here, as that's not what we are testing for. 
Same in other functions

File: app/src/androidTest/java/com/android/periodpals/ui/profile/CreateProfileTest.kt
Comment (charliemangano) [inline app/src/androidTest/java/com/android/periodpals/ui/profile/CreateProfileTest.kt:1]: As in the `MapScreenTest` file, setting the theme is not necessary. Plus here the theme used is the Material one and not our `PeriodPalsAppTheme`.
I saw this file has not been cleaned up yet so I don't know that it's your job to fix this?

File: app/src/main/java/com/android/periodpals/ui/map/Map.kt [hunk 6]
```diff
@@ -50,18 +51,14 @@ fun MapScreen(
...
             onTabSelect = { route -> navigationActions.navigateTo(route) },
             tabList = LIST_TOP_LEVEL_DESTINATION,
             selectedItem = navigationActions.currentRoute())
       },
-      topBar = {
-        TopAppBar(
-            title = "Map",
-        )
-      },
+      topBar = { TopAppBar(title = "Map") },
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/map/Map.kt:61]: "Map" should be extracted to a `private const val SCREEN_TITLE` for better maintainability and readability.

File: app/src/main/java/com/android/periodpals/ui/alert/AlertsList.kt [hunk 7]
```diff
@@ -49,7 +49,7 @@ fun AlertListScreen(navigationActions: NavigationActions) {
       },
       topBar = {
         Column(modifier = Modifier.fillMaxWidth()) {
-          TopAppBar(title = "Alerts List")
+          TopAppBar(title = "Alert Lists")
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertsList.kt:52]: Thanks for changing this! 
It should be extracted to a `private const val SCREEN_TITLE` for better maintainabilty.

File: app/src/main/java/com/android/periodpals/ui/profile/CreateProfile.kt [hunk 8]
```diff
@@ -73,6 +74,7 @@ fun CreateProfileScreen(navigationActions: NavigationActions) {
 
   Scaffold(
       modifier = Modifier.fillMaxSize().testTag(CreateProfileScreen.SCREEN),
+      topBar = { TopAppBar(title = "Create Your Account") },
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/profile/CreateProfile.kt:77]: This should be extracted to a `private const val` as well

File: app/src/main/java/com/android/periodpals/ui/profile/ProfileScreen.kt [hunk 9]
```diff
@@ -70,7 +70,7 @@ fun ProfileScreen(navigationActions: NavigationActions) {
       },
       topBar = {
         TopAppBar(
-            title = "Profile",
+            title = "Your Profile",
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/profile/ProfileScreen.kt:73]: This should be extracted to a `private const val` as well

File: app/src/main/java/com/android/periodpals/ui/timer/Timer.kt [hunk 10]
```diff
@@ -15,16 +17,13 @@ import com.android.periodpals.ui.navigation.TopAppBar
...
                 onTabSelect = { route -> navigationActions.navigateTo(route) },
                 tabList = LIST_TOP_LEVEL_DESTINATION,
                 selectedItem = navigationActions.currentRoute())
           }),
-      topBar = {
-        TopAppBar(
-            title = "Timer",
-        )
-      },
+      topBar = { TopAppBar(title = "Tampon Timer") },
```
Comment (charliemangano) [inline app/src/main/java/com/android/periodpals/ui/timer/Timer.kt:27]: This should be extracted to a `private const val` as well

**Review by charliemangano — CHANGES_REQUESTED** [review]
## Summary
Nice job! I left some notes in the comments.
## Important 
#### Code Quality
Good work making the app consistent with the Figma! Please refactor the screen titles to make the code more maintainable and consistent (see comments)
#### Functionality
Code is correct
## Testing
Tests pass and are complete
## Documentation
See comments
## Suggestions
See comments
## Steps before PR approved
See comments

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt (hunk 1 above)
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:30]: Fixed !

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt (hunk 2 above)
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:3]: Sure !

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt (hunk 3 above)
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:52]: Removed and renamed !

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt (hunk 4 above)
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:63]: Removed and renamed !

File: app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt (hunk 5 above)
Comment (francelu) [reply app/src/androidTest/java/com/android/periodpals/ui/map/MapScreenTest.kt:48]: Removed unnecessary call to `PerioPalsAppTheme`.

**Review by charliemangano — APPROVED** [review]
LGMT! Thank you for implementing the changes!

**Pull request merged**